spec/public-api.yaml

```
@@ -265,6 +265,12 @@ components:
           description:
             If the detector's prediction is below this confidence threshold,
             send the image query for human review.
+        metadata:
+          type: object 
+          nullable: true
+          description:
+            A dictionary of custom key/value metadata to associate with the detector
+            (limited to 1KB).
```

Just making it clear that it's a string representation of a dict, not a dict directly. (And similar below)

```suggestion
            A dictionary of custom key/value metadata to associate with the detector
            (limited to 1KB). This is encoded as a URL-safe, base64-encoded JSON string.
```